sdk/core/core-amqp/src/errors.ts

```diff
@@ -465,114 +465,170 @@ const systemErrorFieldsToCopy: (keyof Omit<NetworkSystemError, "name" | "message
+ * InvalidFieldError: an invalid field was passed in a frame body, and the operation could not proceed.
+ * InvalidOperationError: an operation is attempted but is not allowed.
+ * LinkRedirectError: the address provided cannot be resolved to a terminus at the current container.
+ * MessageLockLostError: the lock on the message is lost.
+ * MessageNotFoundError: message is not found.
+ * MessageTooLargeError: the message sent is too large: the maximum size is 256Kb.
+ * MessageWaitTimeout: no new messages are received for the specified time.
+ * MessagingEntityAlreadyExistsError: an attempt is made to create an entity that already exists.
+ * MessagingEntityDisabledError: trying to access/connect to a disabled messaging entity.
+ * MessagingEntityNotFoundError: the messaging entity does not exist.
+ * NoMatchingSubscriptionError: a matching subscription is not found.
+ * NotImplementedError: a feature is not implemented yet but the placeholder is present.
+ * OperationCancelledError: server cancels the operation due to an internal issue.
+ * OperationTimeoutError: the service fails to respond within a given timeframe.
+ * PartitionNotOwnedError: an attempt is made to access a partition that is not owned by the requesting entity.
+ * PreconditionFailedError: a condition that should have been met in order to execute an operation was not.
+ * PublisherRevokedError: access to publisher has been revoked.
+ * QuotaExceededError: The the Azure EventHub/ServiceBus quota has been exceeded.
+ * ReceiverDisconnectedError: two or more instances connect to the same partition with different epoch values.
+ * RelayNotFoundError: relay is not found.
+ * ResourceDeletedError: a server entity the client is working with has been deleted.
+ * ResourceLockedError: the client attempted to work with a server entity to which it has no access because another client is working with it.
+ * SenderBusyError: the client sender does not have enough link credits to send the message.
+ * ServerBusyError: the server is busy. Callers should wait a while and retry the operation.
+ * ServiceCommunicationError: Error for signaling general communication errors related to messaging operations.
+ * ServiceUnavailableError: the service is unavailable. The operation should be retried.
+ * SessionCannotBeLockedError: the Azure ServiceBus session cannot be locked.
+ * SessionLockLostError: the lock on the Azure ServiceBus session is lost.
+ * SessionWindowViolationError: the peer violated incoming window for the session.
+ * StoreLockLostError: the store lock is lost.
+ * SystemError: a low level system error is thrown by node.js. See {@link https://nodejs.org/api/errors.html#errors_class_systemerror}.
+ * TransferLimitExceededError: the peer sent more message transfers than currently allowed on the link.
+ * UnattachedHandleError: currently in use of an attached link.
+ * UnauthorizedError: the connection parameters are wrong and the server refused the connection.
```

Comment:
Oh dear lord...
Is this to help the case when people hover over the type in the editor?
Because if one is on the docs page, they would see the different types with their own descriptions.

Reply:
Yes, this is for users to document what values are available. :)

These were the descriptions in the file already so if they're inaccurate we should fix them. Where is the canonical source we should use here?

Reply:
I'll sync these descriptions in a really near-term PR.

Reply:
Leaving this for now. We can discuss enum'ing or something else in the next PR.

Reply:
Can fix as part of #12117